core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala

```diff
@@ -727,6 +798,7 @@ object SnappyStoreHiveCatalog {
 
   private[this] var relationDestroyVersion = 0
   private val relationDestroyLock = new ReentrantReadWriteLock()
+  private val alterTableLock = new Object
```

Comment:
Well, may be there are two thread trying to create index at the same time and one overwrites the index of another.